server/src/test/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutorTests.java

```diff
@@ -387,4 +455,15 @@ long getQueueTimeNanos() {
             return queuedTimeTakenNanos;
         }
     }
+
+    /**
+     * Ensures that the time reported by {@code System.nanoTime()} has advanced. It is otherwise feasible for the clock to report no time
+     * passing between operations. Call this method if time passing must be guaranteed.
+     */
+    private static void waitForTimeToElapse() throws InterruptedException {
+        final var startNanoTime = System.nanoTime();
+        while (TimeUnit.MILLISECONDS.convert(System.nanoTime() - startNanoTime, TimeUnit.NANOSECONDS) <= 100) {
+            Thread.sleep(100);
```

Comment:
I think the test will be fine if the clock advances at all, even by 1ns - no need to keep retrying until it hits 100,000,000ns :)

Reply:
Updated to an itty-bitty single nano :) Not sure I'll ever write such a small number again.